### ISSUE TITLE
"xxx is not authorized to user ," with local docker-compose config

### ISSUE THREAD

**Issue by jdespraz** [authelia/authelia#1550]
Hello Authelia Team,

Thanks for the great work! Authelia is a very nice tool and I am considering deploying it in production on my machines.
Unfortunately, I am running into troubles when trying to test a very simple example configuration locally on my machine. For 2 days now, I am constantly running into an error that triggers a redirection to the login page. After a successful login, I get the message:
```Access to https://secure.localhost/ is not authorized to user , redirecting to https://authelia.localhost?rd=https%3A%2F%2Fsecure.localhost%2F```
and I fall back on login.localhost.

I saw similar issues but I don't think they are related and the problem seems to come from something else. Surprisingly, using the examples from https://github.com/authelia/authelia/tree/master/compose straight out-of-the-box give me the error as well...

I was thinking it might be related to the fact that, in this local configuration, my ssl certs are self-signed by traefik and a security feature of firefox might be preventing the redirection somehow (btw this happens on chromium as well). I'm getting out of ideas to resolve this error...

Do you have any hints?

Here is the log for the configuration corresponding to https://github.com/authelia/authelia/tree/master/compose/local). I try to access secure.localhost that should be behind one_factor protection:
```authelia    | time="2020-12-19T02:25:06+11:00" level=info msg="Logging severity set to debug"
authelia    | time="2020-12-19T02:25:06+11:00" level=debug msg="Storage schema is being checked to verify it is up to date"
authelia    | time="2020-12-19T02:25:06+11:00" level=debug msg="Storage schema is up to date"
authelia    | time="2020-12-19T02:25:06+11:00" level=info msg="Authelia is listening for non-TLS connections on 0.0.0.0:9091"
public      | Starting up on port 80
secure      | Starting up on port 80
traefik     | time="2020-12-18T15:25:06Z" level=info msg="Configuration loaded from flags."
authelia    | time="2020-12-19T02:25:48+11:00" level=debug msg="Mark authentication attempt made by user authelia" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:48+11:00" level=debug msg="Credentials validation of user authelia is ok" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:48+11:00" level=debug msg="Required level for the URL https://secure.localhost/ is 1" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:48+11:00" level=debug msg="Redirection URL https://secure.localhost/ is safe" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:49+11:00" level=info msg="Access to https://secure.localhost/ is not authorized to user , redirecting to https://authelia.localhost?rd=https%3A%2F%2Fsecure.localhost%2F" method=GET path=/api/verify remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:57+11:00" level=debug msg="Mark authentication attempt made by user authelia" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:57+11:00" level=debug msg="Credentials validation of user authelia is ok" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:57+11:00" level=debug msg="Required level for the URL https://secure.localhost/ is 1" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:57+11:00" level=debug msg="Redirection URL https://secure.localhost/ is safe" method=POST path=/api/firstfactor remote_ip=172.19.0.1
authelia    | time="2020-12-19T02:25:57+11:00" level=info msg="Access to https://secure.localhost/ is not authorized to user , redirecting to https://authelia.localhost?rd=https%3A%2F%2Fsecure.localhost%2F" method=GET path=/api/verify remote_ip=172.19.0.1
```

Authelia configuration file:
```
###############################################################
#                   Authelia configuration                    #
###############################################################

host: 0.0.0.0
port: 9091
log_level: debug
jwt_secret: a_very_important_secret
default_redirection_url: https://public.localhost
totp:
  issuer: authelia.com

authentication_backend:
  file:
    path: /config/users_database.yml

access_control:
  default_policy: deny
  rules:
    - domain: public.localhost
      policy: bypass
    - domain: traefik.localhost
      policy: one_factor
    - domain: secure.localhost
      policy: one_factor

session:
  name: authelia_session
  secret: unsecure_session_secret
  expiration: 3600 # 1 hour
  inactivity: 300 # 5 minutes
  domain: localhost # Should match whatever your root protected domain is

regulation:
  max_retries: 3
  find_time: 120
  ban_time: 300

storage:
  local:
    path: /config/db.sqlite3

notifier:
  filesystem:
    filename: /config/notification.txt
```

docker-compose file:
```
version: '3.3'

networks:
  net:
    driver: bridge

services:
  authelia:
    image: authelia/authelia
    container_name: authelia
    volumes:
      - ./authelia:/config
    networks:
      - net
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.authelia.rule=Host(`authelia.localhost`)'
      - 'traefik.http.routers.authelia.entrypoints=https'
      - 'traefik.http.routers.authelia.tls=true'
      - 'traefik.http.routers.authelia.tls.options=default'
      - 'traefik.http.middlewares.authelia.forwardauth.address=http://authelia:9091/api/verify?rd=https://authelia.localhost'
      - 'traefik.http.middlewares.authelia.forwardauth.trustForwardHeader=true'
      - 'traefik.http.middlewares.authelia.forwardauth.authResponseHeaders=Remote-User, Remote-Groups, Remote-Name, Remote-Email'
    expose:
      - 9091
    restart: unless-stopped
    environment:
      - TZ=Australia/Melbourne

  traefik:
    image: traefik:v2.2.0
    container_name: traefik
    volumes:
      - ./traefik:/etc/traefik
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - net
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.api.rule=Host(`traefik.localhost`)'
      - 'traefik.http.routers.api.entrypoints=https'
      - 'traefik.http.routers.api.service=api@internal'
      - 'traefik.http.routers.api.tls=true'
      - 'traefik.http.routers.api.tls.options=default'
      - 'traefik.http.routers.api.middlewares=authelia@docker'
    ports:
      - 80:80
      - 443:443
    command:
      - '--api'
      - '--providers.docker=true'
      - '--providers.docker.exposedByDefault=false'
      - '--providers.file.filename=/etc/traefik/certificates.yml'
      - '--entrypoints.http=true'
      - '--entrypoints.http.address=:80'
      - '--entrypoints.http.http.redirections.entrypoint.to=https'
      - '--entrypoints.http.http.redirections.entrypoint.scheme=https'
      - '--entrypoints.https=true'
      - '--entrypoints.https.address=:443'
      - '--log=true'
      - '--log.level=DEBUG'
      - '--log.filepath=/var/log/traefik.log'

  secure:
    image: containous/whoami
    container_name: secure
    networks:
      - net
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.secure.rule=Host(`secure.localhost`)'
      - 'traefik.http.routers.secure.entrypoints=https'
      - 'traefik.http.routers.secure.tls=true'
      - 'traefik.http.routers.secure.tls.options=default'
      - 'traefik.http.routers.secure.middlewares=authelia@docker'
    expose:
      - 80
    restart: unless-stopped

  public:
    image: containous/whoami
    container_name: public
    networks:
      - net
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.public.rule=Host(`public.localhost`)'
      - 'traefik.http.routers.public.entrypoints=https'
      - 'traefik.http.routers.public.tls=true'
      - 'traefik.http.routers.public.tls.options=default'
      - 'traefik.http.routers.public.middlewares=authelia@docker'
    expose:
      - 80
    restart: unless-stopped
```

**Comment by nightah** [comment]
The self-signed certificates are a red herring, this issue is because of the domain you have selected.
More specifically we cannot set a cookie for the root domain `localhost` and your session is being re-generated.

The following stackoverflow post gives you a pretty good write up: https://stackoverflow.com/a/32210291.

If you set your domain to something like the default (example.com) that's used in our `setup.sh` in our `local` compose examples then you won't have any issues.

**Comment by jdespraz** [comment]
Thanks a lot @nightah for the explanation.
I was able to perform my tests without any troubles.

If anyone runs into the same problem, here is the simple workaround I used:
instead of ```localhost```, use the domain ```127.0.0.1.xip.io``` which fakes a domain that points to localhost and for which cookies can be set with Authelia (more info in http://xip.io). Another similar website is https://nip.io/.
